conjur/init_controller.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+
+"""
+InitController module
+
+This module is the controller that facilitates all init actions
+required to successfully configure the conjurrc
+"""
+
+# Builtin
+import logging
+import os
+import sys
+
+from urllib.parse import urlparse
+
+import OpenSSL
+
+class InitController:
+    """
+    InitController
+
+    This class represents the Presentation Layer for the init command
+    """
+    DEFAULT_CONFIG_FILE = os.path.expanduser(os.path.join('~', '.conjurrc'))
+
+    force_overwrite = False
+    init_input_obj = None
+    init_logic = None
+
+    def __init__(self, init_input_obj, init_logic, force):
+        self.init_input_obj = init_input_obj
+        self.init_logic = init_logic
+        self.force_overwrite = force
+
+    def call(self):
+        """
+        Method that facilitates all method calls in this class
+        """
+        fetched_certificate = self.get_server_certificate(self.init_logic,
+                                                          self.init_input_obj)
+        self.get_host_info(self.init_input_obj)
+        self.write_certificate(self.init_logic,
+                               self.init_input_obj,
+                               self.init_input_obj.cert_file,
+                               fetched_certificate)
+        self.init_logic.write_conjurrc(self.DEFAULT_CONFIG_FILE,
+                                       self.init_input_obj,
+                                       self.force_overwrite)
+
+    # pylint: disable=line-too-long
+    @staticmethod
+    def get_server_certificate(init_logic, input_obj):
+        """
+        Method to get the certificate from the Conjur endpoint detailed by the user
+        """
+        url = input_obj.appliance_url if input_obj.appliance_url else input("Enter the URL of your Conjur service: ")
+        input_obj.appliance_url = ''.join(url)
+
+        url = urlparse(input_obj.appliance_url)
+
+        # Fetch certificate information if connection is expected over TLS
+        certificate_path = input_obj.cert_file if input_obj.cert_file else None
+        if certificate_path is not None:
+            input_obj.cert_file = ''.join(certificate_path)
```

Comment:
Why is the join needed?

Reply:
see my comment [above](https://github.com/cyberark/conjur-api-python3/pull/113#discussion_r544839218)

Reply:
Yeah this is an indicator of a bigger problem. Your configuration object should already have objects in the right format

Reply:
see my [comment above](https://github.com/cyberark/conjur-api-python3/pull/113#discussion_r546603699)